### PR TITLE
HDDS-2073. Make SCMSecurityProtocol message based

### PR DESCRIPTION
**Pull request by elek**
Based on HDDS-2067

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 130 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ HDDS-2067 Compile Tests _ |
| 0 | mvndep | 90 | Maven dependency ordering for branch |
| -1 | mvninstall | 51 | hadoop-hdds in HDDS-2067 failed. |
| -1 | mvninstall | 41 | hadoop-ozone in HDDS-2067 failed. |
| -1 | compile | 44 | hadoop-hdds in HDDS-2067 failed. |
| -1 | compile | 25 | hadoop-ozone in HDDS-2067 failed. |
| -0 | checkstyle | 52 | The patch fails to run checkstyle in hadoop-ozone |
| +1 | mvnsite | 0 | HDDS-2067 passed |
| +1 | shadedclient | 1199 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 36 | hadoop-hdds in HDDS-2067 failed. |
| -1 | javadoc | 65 | hadoop-ozone in HDDS-2067 failed. |
| 0 | spotbugs | 1414 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 72 | hadoop-hdds in HDDS-2067 failed. |
| -1 | findbugs | 34 | hadoop-ozone in HDDS-2067 failed. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 61 | Maven dependency ordering for patch |
| -1 | mvninstall | 43 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 30 | hadoop-ozone in the patch failed. |
| -1 | compile | 20 | hadoop-hdds in the patch failed. |
| -1 | compile | 14 | hadoop-ozone in the patch failed. |
| -1 | cc | 20 | hadoop-hdds in the patch failed. |
| -1 | cc | 14 | hadoop-ozone in the patch failed. |
| -1 | javac | 20 | hadoop-hdds in the patch failed. |
| -1 | javac | 14 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 23 | The patch fails to run checkstyle in hadoop-ozone |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 715 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 22 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 20 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 31 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 20 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 28 | hadoop-hdds in the patch failed. |
| -1 | unit | 23 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 3201 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1521 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux da81b0b41866 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | HDDS-2067 / 63c730a |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/branch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out//home/jenkins/jenkins-slave/workspace/hadoop-multibranch_PR-1521/out/maven-branch-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-compile-hadoop-ozone.txt |
| cc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-compile-hadoop-hdds.txt |
| cc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out//home/jenkins/jenkins-slave/workspace/hadoop-multibranch_PR-1521/out/maven-patch-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/testReport/ |
| Max. process+thread count | 412 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/server-scm hadoop-ozone/insight U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 76 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 69 | Maven dependency ordering for branch |
| -1 | mvninstall | 30 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 29 | hadoop-ozone in trunk failed. |
| -1 | compile | 21 | hadoop-hdds in trunk failed. |
| -1 | compile | 16 | hadoop-ozone in trunk failed. |
| -0 | checkstyle | 36 | The patch fails to run checkstyle in hadoop-ozone |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 859 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 22 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 21 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 958 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 31 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 20 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 30 | Maven dependency ordering for patch |
| -1 | mvninstall | 34 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 28 | hadoop-ozone in the patch failed. |
| -1 | compile | 24 | hadoop-hdds in the patch failed. |
| -1 | compile | 19 | hadoop-ozone in the patch failed. |
| -1 | cc | 24 | hadoop-hdds in the patch failed. |
| -1 | cc | 19 | hadoop-ozone in the patch failed. |
| -1 | javac | 24 | hadoop-hdds in the patch failed. |
| -1 | javac | 19 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 29 | The patch fails to run checkstyle in hadoop-ozone |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 711 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 21 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 20 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 31 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 20 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 28 | hadoop-hdds in the patch failed. |
| -1 | unit | 23 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 2463 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1521 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux f393376be269 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1a2a352 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/branch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out//home/jenkins/jenkins-slave/workspace/hadoop-multibranch_PR-1521/out/maven-branch-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-compile-hadoop-ozone.txt |
| cc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-compile-hadoop-hdds.txt |
| cc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out//home/jenkins/jenkins-slave/workspace/hadoop-multibranch_PR-1521/out/maven-patch-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/testReport/ |
| Max. process+thread count | 413 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/server-scm hadoop-ozone/insight U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1521/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/common/src/main/proto/SCMSecurityProtocol.proto [hunk 1]
```diff
@@ -30,33 +30,73 @@ option java_generic_services = true;
 
 option java_generate_equals_and_hash = true;
 
-package hadoop.hdds;
+package hadoop.hdds.security;
 
 import "hdds.proto";
 
+
+message SCMSecurityRequest {
```
Comment (arp7) [inline hadoop-hdds/common/src/main/proto/SCMSecurityProtocol.proto]: A short comment here would be helpful for maintainers.

**Comment by anuengineer** [comment]
@arp7  Thanks for the review. I have rebased and added the comment that Arpit wanted while committing this patch. @elek  Thank you very much for the contribution.

**Pull request closed without merging**